### DISCUSSION TITLE
Energy Deposited by Individual Secondary Charged Particles in Specified Regions

### DISCUSSION THREAD

**Discussion by jstask0** [nrc-cnrc/EGSnrc#1042] [Questions]
I'm trying to create a custom ausgab object, and I'm not sure where to go now that I have the basic skeleton. I want to have the simulation print to a file how much energy each secondary electron deposited to certain regions. Something like "Region1 Energy1, Region2 Energy2" meaning that some secondary electron deposited Energy1 in Region1 and Energy2 in Region2. I've noticed that there's a method for flagging secondaries in the FluenceScoring ausgab object. Beyond that, I'm not sure how to actually record the energy deposited by that secondary and how I would add that information to a separate file. Any help for how to start off would be great. Thanks!

**Comment by ftessier** [comment]
One way would be to create your own scoring array and then accumulate results (whatever they are) based on conditions (history, region, charge, iarg, etc.). But indeed you'll need to flag secondaries. If in `egs_chamber` you can use the `LATCH` variable for that purpose, associating a latch with the primary, but never to secondaries. I don't know what else to suggest without looking at the code. Perhaps share your source code (if you can and want to) so we can discuss in concrete terms.

**Reply by jstask0** [reply]
Thanks for then suggestions. I'll be going out of town soon and plan to work on it more then. I'll try to share what I have when I return.
